langchain/src/chains/combine_docs_chain.ts

```diff
@@ -154,31 +156,40 @@ export class MapReduceDocumentsChain
 
     let currentDocs = docs as Document[];
 
-    for (let i = 0; i < this.maxIterations; i += 1) {
+    let iterationCount = 0;
+    while (iterationCount < SANITY_CHECK_MAX_ITERATION_COUNT) {
+      if (iterationCount >= this.maxIterations) {
+        break;
+      }
+
       const inputs = currentDocs.map((d) => ({
         [this.documentVariableName]: d.pageContent,
         ...rest,
       }));
-      const promises = inputs.map(async (i) => {
+      const results = await this.llmChain.apply(inputs);
```

Comment:
I think we'd want to have the new behaviour behind a flag? Some people might prefer to save some money and not run the map chain if not needed

Reply:
@nfcampos ok I'll do another implem with a flag.